### PR TITLE
feat(components): Update Page component to accept a titleMetaData component [JOB-97759]

### PR DESCRIPTION
**Pull request by jlapstra**
## Motivations

The Campaigns team requires a status label to be present beside the Heading within a page component. Designs can be found [here](https://www.figma.com/design/DjVmtJjQKATkgC5UxFAleU/Automated-Campaigns-v1?node-id=759-135655&t=KOm5iwuUkaZNYCb8-4). 

This could have been implemented without using the Page component, but we determined that it would be beneficial to allow a custom component to be appended to the end of the title. Then, other teams can take advantage of the change, and that the campaigns component using this implementation will maintain consistency with the design system over time (eg: in the case that the Page component style for the design system were to change).

### Added

- Updated the Page component to accept an optional ReactNode to display beside the title of the Page.
- Added a new story for all title related fields

#### Implementation Details
At first, I was hopeful that I could use the `Flex` component, however, this did not allow for the elements to wrap to the next row on small screen sizes.
Next, I attempted to use the `Box` component, but that kept both elements on the same line, but wrapped the individual components, which did not look visually pleasing. 

This is what it looked like using `Box`
Screenshot 2024-06-19 at 17 05 41

I settled for using custom CSS to achieve the desired responsiveness, and style.

Screenshot 2024-06-19 at 15 57 05

## Testing

Testing can be achieved by navigating to the story `WithAdditionalTitleFields`.
It is also possible to follow local setup steps and test this out with JO. See directions [here](https://atlantis.getjobber.com/?path=/docs/introduction--page#local-testing).

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Page/Page.tsx [hunk 1]
```diff
@@ -126,7 +134,10 @@ export function Page({
         <Content>
           <div className={titleBarClasses} ref={titleBarRef}>
             <div>
-              <Heading level={1}>{title}</Heading>
+              <Flex template={["shrink", "shrink"]}>
+                <Heading level={1}>{title}</Heading>
+                {titleMetaData}
```
Comment (jlapstra) [inline packages/components/src/Page/Page.tsx:139]: considering making this

```
{titleMetaData && (
  <div classname={styles.titleRow}>
    {titleMetaData}
  </div>
)}
```

**Review by MattDekinder — COMMENTED** [review]
LGTM - I see you went with a display: flex rather than display: inline but that seems to work the same and lets you gap elements! I saw your thread about the spacing below the title so great job in solving that.

I'll leave approval for someone from Atlantis but this will work for our design and the new prop is inspired by Polaris' title component design, as was discussed in `team-atlantis`

File: docs/components/Page/Web.stories.tsx [hunk 2]
```diff
@@ -68,10 +73,20 @@ WithActions.args = {
...
+export const WithIntro = BasicTemplate.bind({});
+WithIntro.args = {
   title: "Notifications",
   subtitle: "Notify me of all the work",
   intro:
     "Improve job completion rates, stop chasing payments, and boost your customer service by automatically communicating with your clients at key points before, during, and after a job. Read more about Notifications by visiting our [Help Center](https://help.getjobber.com/hc/en-us).",
+  externalIntroLinks: true,
+};
+
+export const WithAdditionalTitleFields = BasicTemplate.bind({});
```
Comment (chris-at-jobber) [inline docs/components/Page/Web.stories.tsx:85]: oooooh thank you!

Comment (chris-at-jobber) [reply docs/components/Page/Web.stories.tsx:85]: Would love to see this example with actions so I can check how it looks when the actions and the label wrap!

Comment (chris-at-jobber) [reply docs/components/Page/Web.stories.tsx:85]: Alternatively, if you can drop some examples of how that looks at small and large breakpoints on this comment, I can design review that way 😃

Comment (jlapstra) [reply docs/components/Page/Web.stories.tsx:85]: Screenshot 2024-06-20 at 11 21 19
Screenshot 2024-06-20 at 11 21 11
Screenshot 2024-06-20 at 11 20 59
Screenshot 2024-06-20 at 11 20 49
Screenshot 2024-06-20 at 11 20 10
Screenshot 2024-06-20 at 11 20 01
Screenshot 2024-06-20 at 11 19 51
Screenshot 2024-06-20 at 11 19 44
Screenshot 2024-06-20 at 11 19 32

Comment (chris-at-jobber) [reply docs/components/Page/Web.stories.tsx:85]: Amazing, thanks @jlapstra !

**Comment by jlapstra** [comment]
Hi @scotttjob ! With the requested changes, I was able to create a prerelease and see JO [pass CI](https://app.circleci.com/pipelines/github/GetJobber/Jobber/250536/workflows/00cbd74c-e48f-4e67-9e63-72bb19255bcc) this time around. Can I get a re-review? :)

**Review by scotttjob — APPROVED** [review]
LGTM!

Thanks for the changes you made!

**Pull request merged**